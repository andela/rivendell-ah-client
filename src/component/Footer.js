const Footer = () =>{
    <div>
        <h2>Author's Haven Footer </h2>
    </div>
    };
    export default Footer ;