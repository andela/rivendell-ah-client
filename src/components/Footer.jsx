import React from 'react';
import { Link } from 'react-router-dom';

const Footer = () => (
  <footer>
    <p>
      &copy; Authors Haven
      &nbsp; &nbsp;
      <Link to="/">About Us</Link>
    </p>
  </footer>
);


export default Footer;
