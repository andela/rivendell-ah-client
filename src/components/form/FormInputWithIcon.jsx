import React from 'react';
import { PropTypes } from 'prop-types';

/**
 * a reusable form input component with custom style
 * @param {Object} props contains:
 * {
 * handleChange(func): handle input change
 * value(string): input value
 * type(string): input type
 * iconName(string): name of materialize icon
 * name(string): name of form input field
 * placeholder(string)
 * children(node): jsx
 * @returns {Function} jsx
 * }
 */
const FormInputWithIcon = (props) => {
  const {
    handleChange, value, type, iconName, name,
    children, placeholder, visibilityIcon, visibilityIconClick } = props;
  return (
    <label htmlFor={name}>
      <div onClick={visibilityIconClick} className="form-field">
        <span className="form-icon-wrapper">
          <i className="material-icons form">{iconName}</i>
        </span>
        <input
          onChange={handleChange}
          value={value}
          type={type}
          name={name}
          placeholder={placeholder}
        />
        {visibilityIcon
          && (
            <span className="form-icon-wrapper right">
              <i
                className="material-icons form"
              >
                {visibilityIcon}
              </i>
            </span>
          )
        }
      </div>
      {children}
    </label>
  );
};

FormInputWithIcon.propTypes = {
  handleChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  iconName: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  children: PropTypes.node,
  visibilityIcon: PropTypes.string,
  visibilityIconClick: PropTypes.func,
};

FormInputWithIcon.defaultProps = {
  children: null,
  visibilityIcon: '',
  visibilityIconClick: () => {}
};

export default FormInputWithIcon;
