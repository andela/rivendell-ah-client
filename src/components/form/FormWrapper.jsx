
import React from 'react';
import { PropTypes } from 'prop-types';
import { Form, Container } from 'semantic-ui-react';
import FormHeaderIcon from './FormHeaderIcon';
import AuthFormFooter from './AuthFormFooter';

/**
 * a reusable form wrapper component, that
 * provides custom styles
 * @param {Object} props contains:
 * {
 * handleSubmit(func): handle form submit
 * isLoading(bool): loading indicator
 * formType(string): login, signup etc...
 * iconName(string): name of materialize icon
 * headerText(string): h2 at the top of the form
 * children(node): jsx
 * @returns {Function} jsx
 * }
 */
const FormWrapper = (props) => {
  const {
    handleSubmit, isLoading, children,
    formType, iconName, headerText
  } = props;
  return (
    <div id={formType}>
      <Container>
        <div className="content">
          <div className="form">
            <Form loading={isLoading} onSubmit={handleSubmit}>
              <FormHeaderIcon iconName={iconName} />
              <h2>{headerText}</h2>
              {children}
            </Form>
          </div>
          <AuthFormFooter authType={formType} />
        </div>
      </Container>
    </div>
  );
};

FormWrapper.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  children: PropTypes.node,
  formType: PropTypes.string.isRequired,
  headerText: PropTypes.string.isRequired,
  iconName: PropTypes.string
};

FormWrapper.defaultProps = {
  iconName: '',
  children: null
};

export default FormWrapper;
