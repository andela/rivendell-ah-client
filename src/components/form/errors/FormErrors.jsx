import React from 'react';
import { PropTypes } from 'prop-types';
import FormValidationErrors from './FormValidationErrors';
import ApiValidationErrors from './ApiValidationErrors';


/**
 * All form validations in one place
 * @param {Object} props contains both api validaiton
 * and local form validation alongside the error type(form input)
 * @returns {Function} jsx
 */
const FormErrors = (props) => {
  const { formValidationErrors, apiValidationErrors, errorType } = props;
  return (
    <ul>
      <FormValidationErrors
        errors={formValidationErrors}
        errorType={errorType}
      />
      <ApiValidationErrors
        errors={apiValidationErrors}
        field={errorType}
      />
    </ul>
  );
};

FormErrors.propTypes = {
  formValidationErrors: PropTypes.shape({
    errorCount: PropTypes.number
  }).isRequired,
  apiValidationErrors: PropTypes.shape({
    message: PropTypes.string,
  }).isRequired,
  errorType: PropTypes.string.isRequired,
};

export default FormErrors;
