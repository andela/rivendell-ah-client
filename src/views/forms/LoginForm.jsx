import React from 'react';
import { PropTypes } from 'prop-types';
import { Form, Button } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import FormInputWithIcon from '../../components/form/FormInputWithIcon';
import FormWrapper from '../../components/form/FormWrapper';
import FormErrors from '../../components/form/errors/FormErrors';
import ApiErrorMsg from '../../components/form/errors/ApiErrorMsg';
import SocialLogin from '../../components/SocialLogin';

const LoginForm = (props) => {
  const {
    handleChange, handleSubmit, handleErrMsgDismiss,
    isLoading, apiValidationErrors, displayErrMsg,
    formValidationErrors, formData, socialMedia
  } = props;
  return (
    <FormWrapper
      isLoading={isLoading}
      handleSubmit={handleSubmit}
      iconName="person"
      formType="login"
      headerText="Login"
    >
      <ApiErrorMsg
        apiValidationErrors={apiValidationErrors}
        handleErrMsgDismiss={handleErrMsgDismiss}
        displayErrMsg={displayErrMsg}
      />
      <div id="social-login">
        <SocialLogin socialMedia={socialMedia} />
      </div>
      <Form.Field error={formValidationErrors.email.length > 0}>
        <FormInputWithIcon
          handleChange={handleChange}
          value={formData.email}
          name="email"
          type="email"
          iconName="email"
          placeholder="Email"
        >
          <FormErrors
            apiValidationErrors={apiValidationErrors}
            formValidationErrors={formValidationErrors}
            errorType="email"
          />
        </FormInputWithIcon>
      </Form.Field>
      <Form.Field error={formValidationErrors.password.length > 0}>
        <FormInputWithIcon
          handleChange={handleChange}
          value={formData.password}
          name="password"
          type="password"
          iconName="lock"
          placeholder="Password"
        >
          <FormErrors
            apiValidationErrors={apiValidationErrors}
            formValidationErrors={formValidationErrors}
            errorType="password"
          />
        </FormInputWithIcon>
      </Form.Field>
      <Button size="medium" fluid>Login</Button>
      <Link to="/forget-password" className="forget-password">
        Forgot password?
      </Link>
      <div className="clear-fix">{}</div>
    </FormWrapper>
  );
};

LoginForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleErrMsgDismiss: PropTypes.func.isRequired,
  apiValidationErrors: PropTypes.shape({
    message: PropTypes.string,
  }).isRequired,
  displayErrMsg: PropTypes.bool.isRequired,
  formValidationErrors: PropTypes.shape({
    errorCount: PropTypes.number.isRequired,
  }).isRequired,
  formData: PropTypes.shape({
    email: PropTypes.string.isRequired
  }).isRequired,
  isLoading: PropTypes.bool.isRequired,
  socialMedia: PropTypes.array.isRequired
};

export default LoginForm;
