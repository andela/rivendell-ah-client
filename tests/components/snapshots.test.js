import { shallow } from 'enzyme';
import React from 'react';
import FormInputWithIcon from '../../src/components/form/FormInputWithIcon';
import Footer from '../../src/components/Footer';
import FormWrapper from '../../src/components/form/FormWrapper';
import FormErrors from '../../src/components/form/errors/FormErrors';
import LoginForm from '../../src/views/forms/LoginForm';
import SignuForm from '../../src/views/forms/SignupForm';

describe('The Form Input With Icon component', () => {
  it('Should render as expected', () => {
    const props = {
      handleChange: () => {},
      value: "validemail@gmail.com",
      type: "email",
      iconName: "email",
      name: "email",
    };
    // const output = FormInputWithIcon(props).toJson();
    const tree = shallow(<FormInputWithIcon
        handleChange={props.handleChange}
        value={props.value}
        type={props.type}
        iconName={props.iconName}
        name={props.name}
      />);
    expect(tree).toMatchSnapshot();
  });
});

describe('The Footer component', () => {
  it('Should render as expected', () => {
    const footer = shallow(<Footer />);
    expect(footer).toMatchSnapshot();
  });
});

describe('The Form Wrapper component', () => {
  it('Should render as expected', () => {
  const props = {
    handleSubmit: () => {},
    isLoading: false,
    formType: "login",
    headerText: "Login",
    iconName: "person",
  }
    const tree = shallow(<FormWrapper
      handleSubmit={props.handleSubmit}
      formType={props.formType}
      isLoading={props.isLoading}
      headerText={props.headerText}
      iconName={props.iconName}
    />);
    expect(tree).toMatchSnapshot();
  });
});

describe('The Form Errors component', () => {
  it('Should render as expected', () => {
    const props = {
      formValidationErrors: {
        errorCount: 0,
      },
      apiValidationErrors: {
        message: '',
      },
      errorType: 'login'
    }
    const tree = shallow(<FormErrors
      formValidationErrors={props.formValidationErrors}
      apiValidationErrors={props.apiValidationErrors}
      errorType={props.errorType}
    />);
    expect(tree).toMatchSnapshot();
  });
});

describe('The Login Form component', () => {
  it('Should render as expected', () => {
    const props = {
      handleChange: () => {},
      handleSubmit: () => {},
      handleErrMsgDismiss: () => {},
      isLoading: false,
      apiValidationErrors: {
        message: '',
        response: {}
      },
      displayErrMsg: true,
      formValidationErrors: {
        errorCount: 0,
        email: [],
        password: []
      },
      formData: {
        email: '',
        password: ''
      }
    }
    const tree = shallow(<LoginForm
      handleChange={props.handleChange}
      handleSubmit={props.handleSubmit}
      handleErrMsgDismiss={props.handleErrMsgDismiss}
      isLoading={props.isLoading}
      apiValidationErrors={props.apiValidationErrors}
      displayErrMsg={props.displayErrMsg}
      formValidationErrors={props.formValidationErrors}
      formData={props.formData}
    />);
    expect(tree).toMatchSnapshot();
  });
});
describe('The Signup Form component', () => {
  it('Should render as expected', () => {
    const props = {
      handleChange: () => {},
      handleSubmit: () => {},
      handleErrMsgDismiss: () => {},
      isLoading: false,
      signUpErrors: {
        message: '',
        response: {}
      },
      displayErrMsg: true,
      validationErrors: {
        errorCount: 0,
        firstName: [],
        lastName: [],
        email: [],
        confirm: [],
        username: [],
        password: []
      },
      formData: {
        email: '',
        password: ''
      }
    }
    const tree = shallow(<SignuForm
      handleChange={props.handleChange}
      handleSubmit={props.handleSubmit}
      handleErrMsgDismiss={props.handleErrMsgDismiss}
      isLoading={props.isLoading}
      signUpErrors={props.signUpErrors}
      displayErrMsg={props.displayErrMsg}
      validationErrors={props.validationErrors}
      formData={props.formData}
    />);
    expect(tree).toMatchSnapshot();
  });
});
