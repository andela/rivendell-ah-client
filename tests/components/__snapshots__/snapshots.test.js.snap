// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Footer component Should render as expected 1`] = `
<footer>
  <p>
    © Authors Haven    
    <Link
      replace={false}
      to="/"
    >
      About Us
    </Link>
  </p>
</footer>
`;

exports[`The Form Errors component Should render as expected 1`] = `
<ul>
  <FormValidationErrors
    errorType="login"
    errors={
      Object {
        "errorCount": 0,
      }
    }
  />
  <ApiValidationErrors
    errors={
      Object {
        "message": "",
      }
    }
    field="login"
  />
</ul>
`;

exports[`The Form Input With Icon component Should render as expected 1`] = `
<label
  htmlFor="email"
>
  <div
    className="form-field"
    onClick={[Function]}
  >
    <span
      className="form-icon-wrapper"
    >
      <i
        className="material-icons form"
      >
        email
      </i>
    </span>
    <input
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="email"
      value="validemail@gmail.com"
    />
  </div>
</label>
`;

exports[`The Form Wrapper component Should render as expected 1`] = `
<div
  id="login"
>
  <Container>
    <div
      className="content"
    >
      <div
        className="form"
      >
        <Form
          as="form"
          loading={false}
          onSubmit={[Function]}
        >
          <FormHeaderIcon
            iconName="person"
          />
          <h2>
            Login
          </h2>
        </Form>
      </div>
      <AuthFormFooter
        authType="login"
      />
    </div>
  </Container>
</div>
`;

exports[`The Login Form component Should render as expected 1`] = `
<FormWrapper
  formType="login"
  handleSubmit={[Function]}
  headerText="Login"
  iconName="person"
  isLoading={false}
>
  <div
    id="social-login"
  >
    <SocialLogin
      socialMedia={Array []}
    />
  </div>
  <div
    className="or-separator"
  >
    <i>
      or
    </i>
  </div>
  <ApiErrorMsg
    apiValidationErrors={
      Object {
        "message": "",
        "response": Object {},
      }
    }
    displayErrMsg={true}
    handleErrMsgDismiss={[Function]}
  />
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="email"
      name="email"
      placeholder="Email"
      type="email"
      value=""
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="email"
        formValidationErrors={
          Object {
            "email": Array [],
            "errorCount": 0,
            "password": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="lock"
      name="password"
      placeholder="Password"
      type="password"
      value=""
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="password"
        formValidationErrors={
          Object {
            "email": Array [],
            "errorCount": 0,
            "password": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <Button
    as="button"
    fluid={true}
    role="button"
    size="medium"
  >
    Login
  </Button>
  <Link
    className="forget-password"
    replace={false}
    to="/forgot-password"
  >
    Forgot password?
  </Link>
  <div
    className="clear-fix"
  />
</FormWrapper>
`;

exports[`The Signup Form component Should render as expected 1`] = `
<FormWrapper
  formType="signup"
  handleSubmit={[Function]}
  headerText="Sign up"
  iconName="person"
  isLoading={false}
>
  <div
    id="social-login"
  >
    <SocialLogin
      socialMedia={Array []}
    />
  </div>
  <div
    className="or-separator"
  >
    <i>
      or
    </i>
  </div>
  <ApiErrorMsg
    apiValidationErrors={
      Object {
        "message": "",
        "response": Object {},
      }
    }
    displayErrMsg={true}
    handleErrMsgDismiss={[Function]}
  />
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="person"
      name="firstName"
      placeholder="First name"
      type="text"
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="firstName"
        formValidationErrors={
          Object {
            "confirm": Array [],
            "email": Array [],
            "errorCount": 0,
            "firstName": Array [],
            "lastName": Array [],
            "password": Array [],
            "username": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="person"
      name="lastName"
      placeholder="Last name"
      type="text"
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="lastName"
        formValidationErrors={
          Object {
            "confirm": Array [],
            "email": Array [],
            "errorCount": 0,
            "firstName": Array [],
            "lastName": Array [],
            "password": Array [],
            "username": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="email"
      name="email"
      placeholder="Email"
      type="email"
      value=""
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="email"
        formValidationErrors={
          Object {
            "confirm": Array [],
            "email": Array [],
            "errorCount": 0,
            "firstName": Array [],
            "lastName": Array [],
            "password": Array [],
            "username": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="person_pin"
      name="username"
      placeholder="Username"
      type="text"
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="username"
        formValidationErrors={
          Object {
            "confirm": Array [],
            "email": Array [],
            "errorCount": 0,
            "firstName": Array [],
            "lastName": Array [],
            "password": Array [],
            "username": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="lock"
      name="password"
      placeholder="Password"
      type="password"
      value=""
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="password"
        formValidationErrors={
          Object {
            "confirm": Array [],
            "email": Array [],
            "errorCount": 0,
            "firstName": Array [],
            "lastName": Array [],
            "password": Array [],
            "username": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <FormField
    error={false}
  >
    <FormInputWithIcon
      handleChange={[Function]}
      iconName="lock"
      name="confirm"
      placeholder="Confirm password"
      type="password"
      visibilityIcon=""
      visibilityIconClick={[Function]}
    >
      <FormErrors
        apiValidationErrors={
          Object {
            "message": "",
            "response": Object {},
          }
        }
        errorType="confirm"
        formValidationErrors={
          Object {
            "confirm": Array [],
            "email": Array [],
            "errorCount": 0,
            "firstName": Array [],
            "lastName": Array [],
            "password": Array [],
            "username": Array [],
          }
        }
      />
    </FormInputWithIcon>
  </FormField>
  <Button
    as="button"
    fluid={true}
    role="button"
    size="medium"
  >
    Sign up
  </Button>
</FormWrapper>
`;
